plugins/node/opentelemetry-instrumentation-dataloader/src/instrumentation.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import {
+  diag,
+  trace,
+  context,
+  Link,
+  SpanStatusCode,
+  SpanKind,
+} from '@opentelemetry/api';
+import { DataloaderInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import * as Dataloader from 'dataloader';
+
+const MODULE_NAME = 'dataloader';
+
+type DataloaderInternal = typeof Dataloader.prototype & {
+  _batchLoadFn: Dataloader.BatchLoadFn<unknown, unknown>;
+  _batch: { spanLinks?: Link[] } | null;
+};
+
+type LoadFn = typeof Dataloader.prototype['load'];
+type LoadManyFn = typeof Dataloader.prototype['loadMany'];
+
+export class DataloaderInstrumentation extends InstrumentationBase<
```

Comment:
I'm pretty sure that this results in consumers need the `Dataloader` type available but they are not in dependencies.
In other instrumentations `InstrumentationBase<any>` was used to avoid this.
If possible we should avoid leaking 3rd party types as this likely ends up in build issues at consumer side.

In the end using `typeof Dataloader` as argument for the generics is mostly a cast of any to `typeof Dataloader` at instrumentation API. You can do the cast also inside the functions to avoid leaking types.

Reply:
Will remove this, thanks!